### DISCUSSION TITLE
where to find a full documentation to guide through data structures/methods

### DISCUSSION THREAD

**Discussion by henriquesati** [mitmproxy/mitmproxy#7246] [General]
Where do I find the necessary documentation to write python scripts? I want to know what methods are avaiable in certains data structures or which properties I can invoke. I couldnt find the type of  documentation I'm used to when coding in other languages, that shows  functions, parameters to pass and the return data type, till now all I could use and write I found on other people's code

**Comment by emanuele-em** [comment]
- custom python scripts in mitmproxy -> addons
- docs -> [here](https://docs.mitmproxy.org/stable/addons-overview/)
- API list -> [here](https://docs.mitmproxy.org/stable/api/events.html)
- Very useful examples with a large list of covered cases -> [here](https://docs.mitmproxy.org/stable/addons-examples/)

happy hacking 🙂
